lightgbm/src/test/scala/com/microsoft/azure/synapse/ml/lightgbm/split1/VerifyLightGBMCommon.scala

```diff
@@ -74,4 +92,206 @@ class VerifyLightGBMCommon extends TestBase {
+    assert(taskDatasetCreationTimes.sum > 0)
+    println(s"Task dataset creation times: ${taskDatasetCreationTimes.mkString(",")}")
+    val taskTrainingIterationTimes = measures.taskTrainingIterationTimes()
+    assert(taskTrainingIterationTimes.nonEmpty)
+    // TODO assert(taskTrainingIterationTimes.sum > 0)
+    println(s"Task training iteration times: ${taskTrainingIterationTimes.mkString(",")}")
+
+    val tasks = measures.getTaskMeasures
+    val activeTasks = tasks.filter(t => t.isActiveTrainingTask).map(t => t.partitionId)
+    println(s"Active task ids: ${activeTasks.mkString(",")}")
+
+    // TODO verify all diff measures that are 0 by default
+  }
+
+  // Utility used for doing local perf testing, so leave ignored unless in use
+  ignore("Performance testing") {
+    // modify this test for getting some simple performance measures
+    val dataset = taskDF
+    val measurementCount = 1
+    val executionModes = Array("bulk")  // streaming, bulk
+    val microBatchSizes = Array(4000) // 1, 2, 4, 8, 16, 32, 100, 1000)
+    val matrixTypes = Array("dense")  // dense, sparse, auto
+    val useSingleDatasetModes = Array(x = true)
+
+    executionModes.foreach(executionMode => {
+      matrixTypes.foreach(matrixType => {
+        microBatchSizes.foreach(microBatchSize => {
+          useSingleDatasetModes.foreach(useSingleDataset => {
+            println(s"*********************************************************************************************")
+            println(s"**** Start ExecutionMode: $executionMode, MatrixType: $matrixType, " +
+              s"useSingleDataset: $useSingleDataset, MicroBatchSize: $microBatchSize")
+            measurePerformance(dataset, measurementCount, executionMode, microBatchSize, matrixType, useSingleDataset)
+            println(s"**** Done ExecutionMode: $executionMode, MatrixType: $matrixType, " +
+              s"useSingleDataset: $useSingleDataset, MicroBatchSize: $microBatchSize")
+            println(s"*********************************************************************************************")
+          })
+        })
+      })
+    })
+  }
+
+  def measurePerformance(df: DataFrame,
+                         measurementCount: Int,
+                         executionMode: String,
+                         microBatchSize: Int,
+                         matrixType: String,
+                         useSingleDataset: Boolean): Unit = {
+    val Array(train, _) = df.randomSplit(Array(0.8, 0.2), seed)
+    val measures = Array.ofDim[InstrumentationMeasures](measurementCount)
+
+    (0 until measurementCount).foreach(i => {
+      val measuredModel = baseModel
+        .setUseSingleDatasetMode(useSingleDataset)
+        .setExecutionMode(executionMode)
+        .setMatrixType(matrixType)
+        .setMicroBatchSize(microBatchSize)
+      println(s"** Start Measurement $i")
+      val _ = measuredModel.fit(train)
+      measures(i) = measuredModel.getPerformanceMeasures.get
```

Comment:
measures should probabblky be accessible on the fit model not the estimator

Reply:
ok, I can think about this as part of the future benchmarking work. The performance measures are already in main, so this PR doesn't really touch that.

I think the decision to put it on the estimator had to do with some interaction between batching, error handling, and the fact that the model doesn't exist until the end. But looking at the code, it looks like we could redesign it that way, and I think you are right.